### ISSUE TITLE
`missing-todo-link` (`TD003`) - false negative when text preceeding "TODO"

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#7628]
```py
# asdf TODO: no error here
# TODO: error
```

**Comment by charliermarsh** [comment]
This may need to be declared a reasonable limitation, because there's always going to be a tradeoff here between false negatives and false positives. If we detect `TD` rules here, it'll be impossible to use `TODO` in prose, like:

```python
# This solves case X.
# We don't solve case Y. That's left as a TODO for future work.
```

The limitation we impose (IIUC) is that the TODO has to be at the start of the comment. I'm not sure we can rely on the colon either (i.e., detect any TODO followed by a colon, even if it's not at the start) because we have rules that detect the lack of colon, so it would be inconsistent in enforcement for those cases.

Any ideas?
